packages/babel-parser/src/types.js

```diff
@@ -1409,6 +1423,16 @@ export type TsNonNullExpression = NodeBase & {
   expression: Expression,
 };
 
+// ================
+// Babel placeholders %%foo%%
+// ================
+
+export type Placeholder<N: PlaceholderTypes> = NodeBase & {
+  type: "Placeholder",
+  id: Identifier,
+  expectedNode: N,
```

Comment:
What do you prefer?
- `expectedNode`
- `expectedType`
- `expected`
- `original`